test/kixi/acceptance/elasticsearch_test.clj

```diff
@@ -1,37 +1,81 @@
 (ns kixi.acceptance.elasticsearch-test
-  (:require [clojure.test :refer :all]
-            [clj-time.core :as t]
-            [environ.core :as env]
-            [kixi.search.elasticsearch :as sut]
+  (:require [clj-time.core :as t]
+            [clojure.spec.test.alpha :as spec-test]
+            [clojure.test :refer :all]
+            [environ.core :refer [env]]
             [kixi.datastore.metadatastore :as md]
+            [kixi.search.elasticsearch :as sut]
             [kixi.search.elasticsearch.event-handlers.metadata-create :as mc]
+            [kixi.spec :refer [alias]]
             [kixi.spec.conformers :as conformers]
-            [kixi.spec :refer [alias]]))
+            [kixi.spec.test-helper :refer [wait-is=]]
+            [kixi.user :as user]
+            [taoensso.timbre :as timbre :refer [error]]))
 
 (alias 'kixi.datastore.metadatastore.query 'mq)
-(alias 'kixi.user 'user)
+
+(timbre/set-level! :warn)
+
+(def wait-tries (Integer/parseInt (env :wait-tries "100")))
+(def wait-per-try (Integer/parseInt (env :wait-per-try "10")))
+
+(defn wait-for-pred
+  ([p]
+   (wait-for-pred p wait-tries))
+  ([p tries]
+   (wait-for-pred p tries wait-per-try))
+  ([p tries ms]
+   (loop [try tries]
+     (if (pos? try)
+       (let [result (p)]
+         (if (not result)
+           (do
+             (Thread/sleep ms)
+             (recur (dec try)))
+           result))
+       (error "Run out of pred attempts!")))))
 
 (defn elasticsearch-url
-  [{:keys [host port]}]
-  (str "http://" host ":" port))
+  [{:keys [protocol host port]}]
+  (str protocol "://" host ":" port))
 
 (defn uuid
   []
   (str (java.util.UUID/randomUUID)))
 
 (def es-url
-  (elasticsearch-url {:host (env/env :elasticsearch-host "localhost")
-                      :port (env/env :elasticsearch-port "9200")}))
+  (elasticsearch-url {:protocol (env :es-protocol "http")
+                      :host (env :es-host "localhost")
+                      :port (env :es-port "9200")}))
+(def es-url "https://vpc-staging-kixi-search-lg6tvmynyrm2ckwwfnntomrxs4.eu-central-1.es.amazonaws.com:443")
```

Comment:
would be nice to get environment specific stuff from the environment

Reply:
ah yeah, this was just for running the tests locally against the staging instance, can bin it now